InterneTaakAfhandeling.Web.Server/Services/OpenKlantApi/OpenKlantApiClient.cs

```
@@ -0,0 +1,119 @@
+            _httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Token",
+                _configuration.GetValue<string>("OpenKlantApi:ApiKey") ?? throw new InvalidOperationException("OpenKlantApi:ApiKey configuration is missing"));
+        }
+
+        public async Task<Actor?> GetActorByEmail(string userEmail)
+        {
+            try
+            {
+                var objectRecords = await _objectApiClient.GetObjectsByEmail(userEmail);
+
+                if (objectRecords == null || objectRecords.Count == 0)
+                    throw new ConflictException("No object records found for the current user.",
+                                                 code: "NO_OBJECT_RECORDS_FOUND");
+
+                if (objectRecords.Count > 1)
+                   throw new ConflictException("Multiple object records found for the current user.",
+                                                 code: "MULTIPLE_OBJECT_RECORDS_FOUND");
+                 
+
+                var response = await _httpClient.GetAsync($"actoren?actoridentificatorObjectId={objectRecords?.FirstOrDefault()?.Data.Identificatie}");
+                response.EnsureSuccessStatusCode();
+
+                var content = await response.Content.ReadFromJsonAsync<ActorResponse>();
+
+                return content?.Results?.FirstOrDefault();
+            }
+            catch (Exception)
+            { 
+                throw;
+            }
+        }
+
+        public async Task<List<Internetaken>> GetInterneTakenByAssignedUser(string? userEmail)
+        {
+            try
+            {
+
+                if (string.IsNullOrEmpty(userEmail))
+                    throw new ConflictException("No email found in the current user's claims.",
+                                                 code: "MISSING_EMAIL_CLAIM");
+
+                var actor = await GetActorByEmail(userEmail) ?? throw new ConflictException("No actor found for the current user.",
```

I don't think we need to throw an exception here, but I'll verify tomorrow. It's possible that a new employee doesn't have any assigned contactverzoeken yet, and then they might not yet have an actor. That's not necessarily a problem. I think we can just return an empty list in that case.

---

Let's indeed return an empty list if there is no actor

---

Comment Addressed